documentation/src/docs/security/securing-distributed-mode-cluster.md

```diff
@@ -0,0 +1,311 @@
+   $ openssl req -new -x509 -keyout ca-key -out ca-cert -days <validity> \
+            -subj "<distinguished_name>" \
+            -passout pass:<strong_password>
+
+   # Sample command
+   $ openssl req -new -x509 -keyout ca-key -out ca-cert -days 365 \
+            -subj "/C=US/ST=Washington/L=Seattle/O=Pravega/OU=CA/CN=Pravega-CA" \
+            -passout pass:changeit
+   ```
+
+2. Create a truststore containing the CA's certificate.
+
+   This truststore will be used by external and internal clients. External clients are client applications connected to a Pravega cluster. Internal clients are services on the server playing the role of a client in internal communications.
+
+   ```
+   $ keytool -keystore client.truststore.jks -noprompt -alias CARoot -import -file ca-cert \
+        -storepass changeit
+
+   # Optionally, list the truststore's contents to verify everything is in order. The output should show
+   # a single entry with alias name `caroot` and entry type `trustedCertEntry`.
+   $ keytool -list -v -keystore client.truststore.jks -storepass changeit
+   ```
+
+That's it! You now have the following CA and client truststore artifacts:
+
+| File | Description |
+|:-----:|:--------|
+| ca-cert | A file containing CA's certificate in .pem format|
+| ca-key |  A password-protected file containing CA's private key  |
+| client.truststore.jks | A password-protected truststore file containing the CA's certificate |
+
+### Stage 2: Obtaining Server Certificates and keys
+
+This phase is doing the following for each service:
+
+1. Generating server certificates and keys
+2. Generating a Certificate Signing Request (CSR)
+3. Submitting the CSR to a CA and obtain a signed certificate
+4. Preparing a keystore containing the signed server certificate and the CA's certificate
+5. Exporting the server certificate's private key
+
+Note:
+
+* For services running on the same host, you may use the same certificate if those services are exposed using the same hostname/IP address. You may even use wildcard certificates to share certificates across hosts. However, it is strongly recommended that you use seperate certificates for each service.
+
+Let's see the steps in detail below.
+
+1. Generate keys and certificates for each service. This certificate is used for establishing TLS as well as for verifying the server's identity.
+
+   ```bash
+   $ keytool -keystore controller01.jks\
+    -genkey -keyalg RSA -keysize 2048 -keypass changeit\
+    -alias controller01 -validity 365\
+    -dname "CN=controller01.pravega.io, OU=Pravega, O=Company, L=Locality, S=StateOrProvince, C=Country"\
+    -ext san=dns:controller01.pravega.io,ip:<IP_ADDRESS>\
+    -storepass changeit
+
+   # Optionally, verify the contents of the generated file:
+   $ keytool -list -v -keystore controller01.jks -storepass changeit
+   ```
+
+2. Generate CSR for each service. It helps to think of a CSR as an application for getting a certificate signed by a trusted authority.
```

Comment:
`It helps to think of a CSR as an application for getting a certificate signed by a trusted authority.`  any rephrasing???

Reply:
I like the imagery in this bit, as it makes it easy to quickly understand what a certificate signing request is.